MerlinAU.sh

```diff
@@ -1652,59 +1652,63 @@ _RunFirmwareUpdateNow_()
         return 1
     fi
 
-    # Extracting the firmware binary image
-    if unzip -o "$FW_ZIP_FPATH" -d "$FW_BIN_DIR" -x README*
-    then
-        rm -f "$FW_ZIP_FPATH"
-    else
-        Say "${REDct}**ERROR**${NOct}: Unable to decompress the firmware ZIP file [$FW_ZIP_FPATH]."
-        "$inMenuMode" && _WaitForEnterKey_ "$menuReturnPromptStr"
-        return 1
-    fi
-
-    # Use Get_Custom_Setting to retrieve the previous choice
-    previous_choice="$(Get_Custom_Setting "local" "n")"
-
-    # Logging user's choice
-    # Check for the presence of "rog" in filenames in the extracted directory
-    cd "$FW_BIN_DIR"
-    rog_file="$(ls | grep -i '_rog_')"
-    pure_file="$(ls | grep -iE '_pureubi.w|_ubi.w' | grep -iv 'rog')"
-
-    local_value="$(Get_Custom_Setting "local")"
-
-if [ -z "$local_value" ]; then
-    if [ -n "$rog_file" ]; then
-        # Check if the first argument is 'run_now'
-        if ! "$inMenuMode" ; then
-            # If the argument is 'run_now' default to the "Pure Build"
-            firmware_file="$pure_file"
-            Update_Custom_Settings "local" "n"
-        else
-            # Otherwise, prompt the user for their choice
-            printf "${REDct}Found ROG build: $rog_file. Would you like to use the ROG build? (y/n)${NOct}\n"
-            read -rp "Enter your choice [$previous_choice]: " choice
-            choice="${choice:-$previous_choice}"
-            if [ "$choice" = "y" ] || [ "$choice" = "Y" ]; then
-                firmware_file="$rog_file"
-                Update_Custom_Settings "local" "y"
-            else
-                firmware_file="$pure_file"
-                Update_Custom_Settings "local" "n"
-            fi
-        fi
-    else
-        firmware_file="$pure_file"
-        Update_Custom_Settings "local" "n"
-    fi
-else
-	# On subsequent runs, use the stored choice without prompting
-    if [ "$previous_choice" = "y" ]; then
-        firmware_file="$rog_file"
-    else
-        firmware_file="$pure_file"
-    fi
-fi
+	# Extracting the firmware binary image
+	if unzip -o "$FW_ZIP_FPATH" -d "$FW_BIN_DIR" -x README*
+	then
+		# Get the default USB mount point path
+		usbMountPoint="$(_GetDefaultUSBMountPoint_)"
+		if [ $? -eq 1 ]; then
+			Say "${REDct}**ERROR**${NOct}: Failed to get the default USB mount point path."
+			# Consider how to handle this error. For now, we'll not delete the ZIP file.
+		else
+			# Check if the ZIP file path is not under the USB mount directory
+			if ! echo "$FW_ZIP_FPATH" | grep -qE "^${usbMountPoint}"
```

Comment:
[Lines 1665-1669] This code results in deleting the ZIP file even when it was on a USB-attached drive (where Entware is installed). I've already got a fix for it.

Reply:
Crap. I was so sure it worked ! :(

Thanks for catching this Martin! This is why I need a second pair of eyes haha. 
2 brains is always better than one.